File: 02-backend/spring-boot-ecommerce/src/main/java/com/shopcart/ecommerce/dto/Purchase.java
package com.shopcart.ecommerce.dto;

import com.shopcart.ecommerce.entity.Address;
import com.shopcart.ecommerce.entity.Customer;
import com.shopcart.ecommerce.entity.Order;
import com.shopcart.ecommerce.entity.OrderItem;
import java.util.Set;
import lombok.Data;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: 03-frontend/angular-ecommerce/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    customer: Customer | undefined;
    shippingAddress!: Address;
    billingAddress!: Address;
    order: Order | undefined;
    orderItems: OrderItem[] | undefined;
}
